mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1751,42 +1751,92 @@ pub fn check_all_utxo_inputs_signed_by_pub(tx: &UtxoTx, expected_pub: &[u8]) ->
 
 pub fn watcher_validate_taker_fee<T: UtxoCommonOps>(
     coin: T,
-    taker_fee_hash: Vec<u8>,
-    verified_pub: Vec<u8>,
+    input: WatcherValidateTakerFeeInput,
+    output_index: usize,
 ) -> ValidatePaymentFut<()> {
+    let expected_amount = input.amount.clone();
+    let sender_pubkey = input.sender_pubkey;
+    let taker_fee_hash = input.taker_fee_hash;
+    let min_block_number = input.min_block_number;
+    let lock_duration = input.lock_duration;
+    let fee_addr = input.fee_addr.to_vec();
+
     let fut = async move {
-        let mut attempts = 0;
-        let taker_fee_hash = H256Json::from(taker_fee_hash.as_slice());
-        loop {
```

Comment:
I had some confusions about the necessity of the loop and decided to remove it, but this might actually be necessary, I'll add it back. The actual reason of retrying the whole validation 4 times is the `get_transaction_bytes` method, right?

Reply:
Here the loop repeats only if `get_transaction_bytes` returns an error:

https://github.com/KomodoPlatform/atomicDEX-API/blob/42c84d0ec92d74b4a4085ee5fe62b4f2360a56f6/mm2src/coins/utxo/utxo_common.rs#L3607-L3656

But here, the whole validation is repeated for any error returned:  

https://github.com/KomodoPlatform/atomicDEX-API/blob/42c84d0ec92d74b4a4085ee5fe62b4f2360a56f6/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L711-L738

Is there a reason for this, or should we make it the same way as the `validate_payment` above? For `watcher_validate_taker_fee` I'll do it the same way as the `validate_payment`.

Reply:
I've noticed that you already returned the loop back) Sorry, missed your question.
Yes, I think it's worth to have the loop here even if we have a loop at the `maker_swap.rs`